Feature/15 create the user context

hitohata opened the pull request:
## Overviews of implementation

Add a user context.
This is the draft of this repository structure.

## Review points

I wrote an application in a user context.
I wrote a 'create user use case' and an 'update user name use case'.
Also, updated the layer and the DDD abstract classes.

taniguchiiqqq commented on backend/main/src/Shared/Utils/Errors.ts:
```diff
@@ -0,0 +1,29 @@
+
+/**
+ * This is base class of use case.
```
Is this comment supposed to be "base class of error"? It's up to you if you modify this comment because it doesn't show in the TypeDoc.

hitohata replied:
This is an old comment. Sorry, I forgot to modify it. The correct comment is the following.
This is the base class error of this application.

taniguchiiqqq commented on backend/main/src/Shared/Layers/UseCase/UseCase.ts:
How about changing this filename to IUseCase?

taniguchiiqqq reviewed (approved):
LGTM with nits.
I just mentioned about minor concerns.

hitohata commented on backend/main/src/User/Domain/UserRepository.ts:
OK, sir.

taniguchiiqqq commented:
I just want to make sure if my understanding is correct. You don't have to answer these questions before merging.
`backend/main/src/Shared/Layers/Controller/Controller.ts`: Does this file belong to the Presentation layer in the DDD context? I guess so because the class in this file interacts with an external system. I mean, it gets events from Lambda.

`backend/main/src/User/Controllers/CreateUserController.ts`: Does this file belong to the Application layer in the DDD context?

hitohata commented:
@taniguchiiqqq 
Thank you for your review and comments.

> backend/main/src/Shared/Layers/Controller/Controller.ts: Does this file belong to the Presentation layer in the DDD context? I guess so because the class in this file interacts with an external system. I mean, it gets events from Lambda.

This controller is the presentation or application layer in the DDD context. I added a Clean Architecture way to this. So, I named it as the controller.
The controller is interacting port with an external system, in this case, lambda input. If the external system was increased, we should add some controllers to this file.

> backend/main/src/User/Controllers/CreateUserController.ts: Does this file belong to the Application layer in the DDD context?

The DDD context can be considered to kind of microservice because it is divided by bounded context.
I think the context should be complete on its own. So, I added this file to this directory. This directory, the User directory, represents the user context. When we add another context, the controller should be added there.

I wrote like above, but this is one of my opinions. Let's brush up on the naming and structure through the development of this project.
